### PR TITLE
Updating charts config for move to helm GH org

### PR DESCRIPTION
**Pull request by mattfarina**
The charts repo is being set to move to the helm org. This config change is to support that.

Note, I believe the account k8s-ci-robot needs to accept the invite to join that org.

### CODE REVIEW

**Comment by krzyzacy** [comment]
I'll probably need to update https://github.com/kubernetes/test-infra/pull/8706 as well, do you want to own charts jobs?

**Comment by mattfarina** [comment]
/hold

Putting a hold on this so that it can be approved and lgtm but we can control about when it's merged for timing purposes.

File: prow/plugins.yaml [hunk 1]
```diff
@@ -211,9 +214,11 @@ plugins:
     - wip
     - trigger
 
-  kubernetes/charts:
+  helm/charts:
```
Comment (cblecker) [inline prow/plugins.yaml]: you'll probably want to copy more of this list: https://github.com/kubernetes/test-infra/blob/72e3821fe1baa8332ae8c78e805049033fd8f5a0/prow/plugins.yaml#L378-L396

**Comment by mattfarina** [comment]
@krzyzacy You could put myself, @viglesiasce, @prydonius, and @unguiculus alongside foxish on that. Do I want to own it? Likely not. But, it someone should be on it.

File: prow/plugins.yaml (hunk 1 above)
Comment (mattfarina) [reply prow/plugins.yaml]: I was just going through that list to make some more additions :)

File: prow/config/tide_test.go [hunk 2]
```diff
@@ -113,6 +113,7 @@ func TestMergeMethod(t *testing.T) {
 		MergeType: map[string]github.PullRequestMergeType{
 			"kubernetes/kops":             github.MergeRebase,
 			"kubernetes/charts":           github.MergeSquash,
+			"helm/charts":                 github.MergeSquash,
```
Comment (stevekuznetsov) [inline prow/config/tide_test.go]: Don't think this should change

File: prow/plugins.yaml [hunk 3]
```diff
@@ -69,6 +68,10 @@ approve:
   - bazelbuild
   implicit_self_approve: true
   review_acts_as_approve: true
+- repos:
```
Comment (stevekuznetsov) [inline prow/plugins.yaml]: Why not just change the name of the repo in the above list?

File: prow/config/tide_test.go (hunk 2 above)
Comment (BenTheElder) [reply prow/config/tide_test.go]: this has to change for the PR to pass, separately this test should be in config/ something since it is kube specific.

Comment (stevekuznetsov) [reply prow/config/tide_test.go]: Really? This test looks like a test for the `MergeMethod()` function off of this test config ... a little confused how this is passing actually with this change but nothing changing on line 138

**Comment by krzyzacy** [comment]
@mattfarina I'll let you rebase and add more people on the OWNERS file :-)

File: prow/config/tide_test.go (hunk 2 above)
Comment (mattfarina) [reply prow/config/tide_test.go]: It appears bazel CI runs are checking the real status of the merge method. So, when I simply changed the "kubernetes/charts" to "helm/charts" things failed. That's why both are here. I expect we can remove the "kubernetes/charts" after the move.

File: prow/plugins.yaml (hunk 3 above)
Comment (mattfarina) [reply prow/plugins.yaml]: I noticed the pattern was to break out those not in the kubernetes org to separate config (e.g., bazelbuild). Even when the config is the same. I was trying to follow the existing convention.

**Comment by mattfarina** [comment]
@krzyzacy Thanks and done.

**Comment by mattfarina** [comment]
If this is good upon review I would like to pair with someone to move the charts repo at the same time we merge this.

File: prow/plugins.yaml (hunk 3 above)
Comment (stevekuznetsov) [reply prow/plugins.yaml]: Seems like an antipattern? I don't feel strongly, defer to engprod

File: prow/config/tide_test.go (hunk 2 above)
Comment (stevekuznetsov) [reply prow/config/tide_test.go]: ew :|

Comment (BenTheElder) [reply prow/config/tide_test.go]: yeah I don't know what is up here, I'd cricle back to reconsider what's going on with this test after this PR, but patching it to pass like this seems like the right thing to do within the scope of this PR

**Comment by mattfarina** [comment]
@cblecker Thanks for bring that up. I'm all over that one. I have a script that will let me audit who is in an owners file, is not on the org, and add them. Right after the move I'll automate making sure it's setup.

File: prow/plugins.yaml (hunk 3 above)
Comment (BenTheElder) [reply prow/plugins.yaml]: breaking them out kinda makes sense, at some point we might want seperate prow deployments? it's sorta iffy what should be on this one, for not it's just easiest to use this one for everything.

**Pull request merged**